src/Simulator/IncrementalRedstoneSimulator/IncrementalRedstoneSimulator.cpp

```diff
@@ -125,8 +125,11 @@ void cIncrementalRedstoneSimulator::Simulate(float a_dt)
 		}
 
 		auto CurrentHandler = cIncrementalRedstoneSimulator::CreateComponent(m_World, CurrentBlock, &m_Data);
-		if (CurrentHandler == nullptr)
+		if (CurrentHandler == nullptr)  // Block at CurrentPosition doesn't have a corresponding redstone handler
 		{
+			// Clean up cached PowerData for CurrentPosition
+			static_cast<cIncrementalRedstoneSimulator *>(m_World.GetRedstoneSimulator())->GetChunkData()->ErasePowerData(CurrentLocation);
```

Comment:
Hm... chunk loading forces this to be called **a lot**, but it's a cheap call and the complexity should be constant. Don't think that will be a problem.